### PR TITLE
feat: refactor project page tabs and funding cycle panel for v4 [1/n]

### PR DESCRIPTION
**Pull request by johnnyd-eth**
- Move version agnostic **project page tabs** and **funding cycle tab** components from `/v2v3/` to `/src/components`. e.g.
     - ConfigurationPanel
    - ConfigurationTable
    - CyclesPanelTooltips
    - CyclesTab
    - UpcomingCycleChangesCallout
    - DisplayCard

- Create `useV2V3UpcomingCycleHasChanges` hook  and refactor `UpcomingCycleChangesCallout` to make it version agnostic (worth)

- Copy and paste version specific components/hooks into V4

**Next step:**
- Get real data in `useV4CycleSection` and `useV4CurrentUpcomingSubPanel`

**Current look:**

Screenshot 2024-07-10 at 12 31 57 pm

### CODE REVIEW

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4CycleSection.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export const useV4CycleSection = (
+  type: 'current' | 'upcoming',
+): ConfigurationPanelTableData => {
+  return {
+    duration: {
+      name: 'Duration',
+      new: '60 days',
+    }
+  }
+  // const { projectId } = useProjectMetadataContext()
```
Comment (johnnyd-eth) [inline src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4CycleSection.ts:12]: Next step

File: src/packages/v4/views/V4ProjectDashboard/hooks/useV4CurrentUpcomingSubPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { t } from '@lingui/macro'
+import { useProjectMetadataContext } from 'contexts/ProjectMetadataContext'
+import { useFundingCycleCountdown } from 'packages/v2v3/components/V2V3Project/ProjectDashboard/hooks/useFundingCycleCountdown'
+import { useProjectContext } from 'packages/v2v3/components/V2V3Project/ProjectDashboard/hooks/useProjectContext'
+import { timeSecondsToDateString } from 'packages/v2v3/components/V2V3Project/ProjectDashboard/utils/timeSecondsToDateString'
+import { useProjectUpcomingFundingCycle } from 'packages/v2v3/hooks/contractReader/useProjectUpcomingFundingCycle'
+import { BallotState } from 'packages/v2v3/models/fundingCycle'
+import { useMemo } from 'react'
+
+export const useV4CurrentUpcomingSubPanel = (type: 'current' | 'upcoming') => {
```
Comment (johnnyd-eth) [inline src/packages/v4/views/V4ProjectDashboard/hooks/useV4CurrentUpcomingSubPanel.tsx:10]: Next step: get V4 data for everything here

File: src/components/Project/ProjectTabs/CyclesPayoutsTab/ConfigurationPanel.tsx [hunk 3]
```diff
@@ -16,9 +16,9 @@ export type ConfigurationPanelTableData = {
 
 type ConfigurationPanelProps = {
   cycle: ConfigurationPanelTableData
-  token: ConfigurationPanelTableData
-  otherRules: ConfigurationPanelTableData
-  extension: ConfigurationPanelTableData | null
+  token?: ConfigurationPanelTableData // TODO: don't make token optional
```
Comment (johnnyd-eth) [inline src/components/Project/ProjectTabs/CyclesPayoutsTab/ConfigurationPanel.tsx:19]: Just so I could reuse this component without needing the other sections

File: src/components/Project/ProjectTabs/CyclesPayoutsTab/UpcomingCycleChangesCallout.tsx [hunk 4]
```diff
@@ -1,27 +1,14 @@
...
-  )
+export const UpcomingCycleChangesCallout = ({
+  hasChanges,
+  loading,
+}: {
+  hasChanges: Boolean,
+  loading: Boolean
+}) => {
   const text = hasChanges
     ? t`This cycle has upcoming changes`
```
Comment (aeolianeth) [inline src/components/Project/ProjectTabs/CyclesPayoutsTab/UpcomingCycleChangesCallout.tsx:13]: This copy will need to change in V4. Maybe we just don't make this component shared? Or make the banner Callout shared but not the copy?

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4UpcomingCycleHasChanges.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { useMemo } from "react"
+import { useV4CurrentUpcomingConfigurationPanel } from "./useV4CurrentUpcomingConfigurationPanel"
+
+export const useV4UpcomingCycleHasChanges = () => {
+  const data = useV4CurrentUpcomingConfigurationPanel('upcoming')
+
+  const loading = useMemo(
```
Comment (aeolianeth) [inline src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4UpcomingCycleHasChanges.ts:7]: In general we can chill out a bit on useMemo. In fact we shouldn't use useMemo until we have a clear performance issue that we need to address with it.

Comment (johnnyd-eth) [reply src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4UpcomingCycleHasChanges.ts:7]: Ok got it haha, this was just copied code.

File: src/components/Project/ProjectTabs/CyclesPayoutsTab/UpcomingCycleChangesCallout.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/components/Project/ProjectTabs/CyclesPayoutsTab/UpcomingCycleChangesCallout.tsx:13]: Got it, will fix this.

**Pull request merged**